shuup/front/basket/__init__.py

```diff
@@ -27,15 +27,15 @@ def get_basket_command_dispatcher(request):
     return cached_load("SHUUP_BASKET_COMMAND_DISPATCHER_SPEC")(request=request)
 
 
-def get_basket(request):
+def get_basket(request, basket_name="basket", basket_class=None):
     """
     :type request: django.http.request.HttpRequest
     :rtype: shuup.front.basket.objects.BaseBasket
     """
-    if not hasattr(request, "basket"):
+    if basket_name == "basket" and hasattr(request, "basket"):
```

Comment:
IMHO this shouldn't do any caching for the basket (to request) anyway. It's the front middleware which is supposed to set `request.basket`. But for backward compatibility it's probably best to not change the existing behavior for the default basket.